src/Microsoft.OData.Core/RawValueWriter.cs

```diff
@@ -86,6 +90,21 @@ public void Dispose()
             this.textWriter = null;
         }
 
+#if NETSTANDARD2_0 || NETCOREAPP3_1_OR_GREATER
+        public ValueTask DisposeAsync()
+        {
+            return DisposeAsyncCore();
+        }
+
+        private async ValueTask DisposeAsyncCore()
+        {
+            Debug.Assert(this.textWriter != null, "The text writer has not been initialized yet.");
+
+            await this.textWriter.FlushAsync().ConfigureAwait(false);

```

Comment:
I think you should explicitly check that `this.textWriter` is not null before you call it.

Ideally, calling `Dispose()` or `DisposeAsync()` multiple times should be idempotent right? In this case calling it the second time will throw a `NullReferenceException`